### DISCUSSION TITLE
How to modify the text of a node from a custom plugin using vue

### DISCUSSION THREAD

**Discussion by n-xt** [ueberdosis/tiptap#2394] [General Questions] [answered]
I created my custom extension which shows a vue component. The vue component has a button which open a modal dialog where the user can enter a new value for the text node. But how can I update the text node? When I use the props, then I get an error that I cannot make changes on prop properties. I want to update the text of "props.node?.firstChild?.text".

**Comment by hanspagel** [comment, marked as answer]
Try getting the position from the Node, setting the text selection, and replace the content with insertContent:

https://tiptap.dev/guide/node-views/vue#get-pos
https://tiptap.dev/api/commands/set-text-selection
https://tiptap.dev/api/commands/insert-content
https://tiptap.dev/api/commands#chain-commands

Hope that helps!

**Reply by n-xt** [reply]
Thanks. I got it working with following code snippet:

```
const updateValue = (val: string) => {
      if (
        props.getPos != undefined &&
        props.deleteNode != undefined &&
        props.editor != undefined
      ) {
        const pos = props.getPos();
        props.deleteNode();
        props.editor
          .chain()
          .setTextSelection(pos)
          .insertContent("<placeholder>" + val + "</placeholder>")
          .run();
      }
    };
```
